python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -351,11 +353,40 @@ cdef _get_np_scalar_from_timedelta64(unique_ptr[scalar]& s):
         raise ValueError("Could not convert cudf::scalar to numpy scalar")
 
 
-def as_scalar(val, dtype=None):
-    if isinstance(val, Scalar):
-        if (dtype is None or dtype == val.dtype):
+def as_device_scalar(val, dtype=None):
+    if dtype:
+        if isinstance(val, (cudf.Scalar, DeviceScalar)):
+            raise TypeError("Can't update dtype of existing GPU scalar")
+        else:
+            return cudf.Scalar(value=val, dtype=dtype).device_value
+    else:
+        if isinstance(val, DeviceScalar):
             return val
+        if isinstance(val, cudf.Scalar):
+            return val.device_value
         else:
-            return Scalar(val.value, dtype)
+            return cudf.Scalar(val).device_value
+
+
+def _is_null_host_scalar(slr):
+    if slr is None or slr is cudf.NA:
+        return True
+    elif isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr):
+        return True
+    else:
+        return False
+
+
+def _create_dummy_nat_scalar(dtype):
```

Comment:
Let's avoid use of this term (https://developers.google.com/style/inclusive-documentation#ableist-language). Does `placeholder` work instead?

Reply:
rolled with `proxy`